### PR TITLE
route_services: add caveat about using ip-restriction service with CDNs

### PR DESCRIPTION
**Pull request by risicle**
What
----

https://www.pivotaltracker.com/story/show/180890489

Thinking about this more, if we were in other circumstances the thing to do here would probably be to just add a script to the `paas-ip-authentication-route-service` repo that would scrape and insert/update the appropriate IPs for them.

How to review
-------------

Describe the steps required to test the changes.

1. Preview the content locally ([see README](https://github.com/alphagov/paas-tech-docs#preview)) and check that it renders as expected.
1. Manually check that any removed or renamed anchor tags are not in use ([linkchecker](https://github.com/linkcheck/linkchecker) doesn't do this).
1. …

### CODE REVIEW

File: source/documentation/deploying_services/route_services.md [hunk 1]
```diff
@@ -139,6 +139,15 @@ This example app uses the `X-Forwarded-For` header added by the GOV.UK PaaS rout
 Using an IP address to authenticate HTTP requests is not sufficient as a standalone authentication mechanism.
 You should use other authentication and authorization methods in your apps.
 
+Note: the example nginx configuration used in this app will not work out-of-the-box
+when used with requests routed via a CDN. This includes requests routed via our
```
Comment (cmcnallygds) [inline source/documentation/deploying_services/route_services.md:143]:
```suggestion
when used with requests routed through a CDN. This includes requests routed through our
```

File: source/documentation/deploying_services/route_services.md [hunk 2]
```diff
@@ -139,6 +139,15 @@ This example app uses the `X-Forwarded-For` header added by the GOV.UK PaaS rout
...
+own AWS CloudFront-based [cdn-route services](/deploying_services/use_a_custom_domain/). For these to work, [`set_real_ip_from`](https://nginx.org/en/docs/http/ngx_http_realip_module.html#set_real_ip_from)
+entries need to be added to `nginx.conf` for all IP-ranges the CDN may forward
```
Comment (cmcnallygds) [inline source/documentation/deploying_services/route_services.md:145]:
```suggestion
entries need to be added to `nginx.conf` for all IP ranges the CDN may forward
```

File: source/documentation/deploying_services/route_services.md [hunk 3]
```diff
@@ -139,6 +139,15 @@ This example app uses the `X-Forwarded-For` header added by the GOV.UK PaaS rout
...
+traffic from. For CloudFront, this includes potentially hundreds of address ranges
+which are [published by AWS](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/LocationsOfEdgeServers.html)
+via a JSON file. These ranges are not completely static and will need to be
```
Comment (cmcnallygds) [inline source/documentation/deploying_services/route_services.md:148]:
```suggestion
via a JSON file. These ranges are not completely static and you will need to 
```

File: source/documentation/deploying_services/route_services.md [hunk 4]
```diff
@@ -139,6 +139,15 @@ This example app uses the `X-Forwarded-For` header added by the GOV.UK PaaS rout
...
+updated occasionally.
```
Comment (cmcnallygds) [inline source/documentation/deploying_services/route_services.md:149]:
```suggestion
update them occasionally.
```

**Review by cmcnallygds — CHANGES_REQUESTED** [review]
Requesting some style changes so content is line with style guide

**Comment by risicle** [comment]
My beautiful hyphens!

File: source/documentation/deploying_services/route_services.md [hunk 5]
```diff
@@ -139,6 +139,15 @@ This example app uses the `X-Forwarded-For` header added by the GOV.UK PaaS rout
 Using an IP address to authenticate HTTP requests is not sufficient as a standalone authentication mechanism.
 You should use other authentication and authorization methods in your apps.
 
+Note: the example nginx configuration used in this app will not work out of the box
+when used with requests routed through a CDN. This includes requests routed through our
+own AWS CloudFront-based [cdn-route services](/deploying_services/use_a_custom_domain/). For these to work, [`set_real_ip_from`](https://nginx.org/en/docs/http/ngx_http_realip_module.html#set_real_ip_from)
```
Comment (cmcnallygds) [inline source/documentation/deploying_services/route_services.md:144]:
```suggestion
own AWS CloudFront-based [cdn-route services](/deploying_services/use_a_custom_domain/). For these to work, you need to add [`set_real_ip_from`](https://nginx.org/en/docs/http/ngx_http_realip_module.html#set_real_ip_from)
```

File: source/documentation/deploying_services/route_services.md [hunk 6]
```diff
@@ -139,6 +139,15 @@ This example app uses the `X-Forwarded-For` header added by the GOV.UK PaaS rout
...
+entries need to be added to `nginx.conf` for all IP ranges the CDN may forward
```
Comment (cmcnallygds) [inline source/documentation/deploying_services/route_services.md:145]:
```suggestion
entries to `nginx.conf` for all IP ranges the CDN may forward
```

Comment (cmcnallygds) [reply source/documentation/deploying_services/route_services.md:145]: Could we avoid the passive voice here by asking the user directly to carry out this action?

**Review by cmcnallygds — CHANGES_REQUESTED** [review]
Suggesting the use of active voice rather than passive

**Comment by risicle** [comment]
All sounds very dictatorial to me but if it's the style guide...

**Review by cmcnallygds — APPROVED** [review]

**Review by cadmiumcat — APPROVED** [review]

**Pull request merged**